### PR TITLE
feat(css_parser): parse @position-try and @view-transition at-rules

### LINKED ISSUE #4685: 📎 Add support for new CSS at-rules
### Description

To enhance our CSS parsing capabilities, we need to add support for two new CSS at-rules
- [@position-try](https://developer.mozilla.org/en-US/docs/Web/CSS/@position-try)
- [@view-transition](https://developer.mozilla.org/en-US/docs/Web/CSS/@view-transition)

You can start around here
https://github.com/biomejs/biome/blob/d80805762cd1efdeaafd2e9d542270bfced8e8f4/xtask/codegen/src/css_kinds_src.rs#L231-L243

### PR DESCRIPTION
**Pull request by kevinmingtarja**
## Summary

Closes #4685.

This PR adds support for parsing these two new CSS at-rules:
[@position-try](https://drafts.csswg.org/css-anchor-position-1/#at-ruledef-position-try)
```
@position-try = @position-try <dashed-ident> { <declaration-list> }
```

[@view-transition](https://drafts.csswg.org/css-view-transitions-2/#view-transition-rule)
```
@view-transition = @view-transition { <declaration-list> }
```

## Test Plan
- Uncommented some lines in `crates/biome_css_analyze/tests/specs/nursery/noUnknownAtRule/valid.css` that were previously failing
- Added unit tests for both at-rules under `crates/biome_css_parser/tests/css_test_suite/ok/at_rule/`

### CODE REVIEW

File: crates/biome_css_formatter/src/css/statements/position_try_at_rule.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+use crate::prelude::*;
+use biome_css_syntax::CssPositionTryAtRule;
+use biome_rowan::AstNode;
+#[derive(Debug, Clone, Default)]
+pub(crate) struct FormatCssPositionTryAtRule;
+impl FormatNodeRule<CssPositionTryAtRule> for FormatCssPositionTryAtRule {
+    fn fmt_fields(&self, node: &CssPositionTryAtRule, f: &mut CssFormatter) -> FormatResult<()> {
+        format_verbatim_node(node.syntax()).fmt(f)
```
Comment (dyc3) [inline crates/biome_css_formatter/src/css/statements/position_try_at_rule.rs:8]: We should format these new nodes. You can take inspiration from how some of the other at-rules are formatted.

File: crates/biome_css_formatter/src/css/statements/view_transition_at_rule.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+use crate::prelude::*;
+use biome_css_syntax::CssViewTransitionAtRule;
+use biome_rowan::AstNode;
+#[derive(Debug, Clone, Default)]
+pub(crate) struct FormatCssViewTransitionAtRule;
+impl FormatNodeRule<CssViewTransitionAtRule> for FormatCssViewTransitionAtRule {
+    fn fmt_fields(&self, node: &CssViewTransitionAtRule, f: &mut CssFormatter) -> FormatResult<()> {
+        format_verbatim_node(node.syntax()).fmt(f)
```
Comment (dyc3) [inline crates/biome_css_formatter/src/css/statements/view_transition_at_rule.rs:8]: Same here, we should format these new nodes.

**Review by dyc3 — CHANGES_REQUESTED** [review]
Looking good! We should add formatting for the new nodes.

File: crates/biome_css_formatter/src/css/statements/position_try_at_rule.rs (hunk 1 above)
Comment (kevinmingtarja) [reply crates/biome_css_formatter/src/css/statements/position_try_at_rule.rs]: Ah right i missed that, i've just added it!

File: crates/biome_css_formatter/src/css/statements/view_transition_at_rule.rs (hunk 2 above)
Comment (kevinmingtarja) [reply crates/biome_css_formatter/src/css/statements/view_transition_at_rule.rs]: Added as well

**Review by dyc3 — APPROVED** [review]
LGTM. Merry Christmas!

**Comment by kevinmingtarja** [comment]
> LGTM. Merry Christmas!

Thank you, merry Christmas to you too!

**Pull request merged**